docs/standard/commandline/syntax.md

```diff
@@ -0,0 +1,572 @@
+
+```dotnetcli
+dotnet add package System.CommandLine --prerelease
+dotnet add package --prerelease System.CommandLine 
+```
+
+Options can be specified in any order. The following commands are equivalent:
+
+```dotnetcli
+dotnet add package System.CommandLine --prerelease --no-restore --source https://api.nuget.org/v3/index.json
+dotnet add package System.CommandLine --source https://api.nuget.org/v3/index.json --no-restore --prerelease 
+```
+
+When there are multiple arguments, the order does matter. The following commands are not equivalent:
+
+```console
+myapp argument1 argument2
+myapp argument2 argument1
+```
+
+## Aliases
+
+In both POSIX and Windows, it's common for some commands and options to have aliases. These are usually short forms that are easier to type. POSIX short forms typically have a single leading hyphen followed by a single character. The following commands are equivalent:
+
+```dotnetcli
+dotnet build --verbosity quiet
+dotnet build -v quiet
+```
+
+The [GNU standard](https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html) recommends automatic aliases. That is, you can enter any part of a long-form command or option name and it will be accepted. This behavior would make the following command lines equivalent:
+
+```dotnetcli
+dotnet publish --output ./publish
+dotnet publish --outpu ./publish
+dotnet publish --outp ./publish
+dotnet publish --out ./publish
+dotnet publish --ou ./publish
+dotnet publish --o ./publish
+```
+
+`System.CommandLine` doesn't support automatic aliases.
+
+## Case sensitivity
+
+Command and option names and aliases are case-sensitive by default according to POSIX convention, and `System.CommandLine` follows this convention. If you want your CLI to be case insensitive, define aliases for the various casing alternatives. For example, `--additional-probing-path` could have aliases `--Additional-Probing-Path` and `--ADDITIONAL-PROBING-PATH`.
+
+In some command-line tools, a difference in casing specifies a difference in function. For example, [`git clean -X`](https://git-scm.com/docs/git-clean#Documentation/git-clean.txt--X) behaves differently than [`git clean -x`](https://git-scm.com/docs/git-clean#Documentation/git-clean.txt--x). The .NET CLI is all lowercase.
+
+## The `--` token
+
+POSIX convention interprets the double-dash (`--`) token as an escape mechanism. Everything to the right of the double-dash token is interpreted as arguments for the command. This functionality can be used to submit arguments that look like options, since it prevents them from being interpreted as options.
+
+Suppose *myapp* takes a `message` argument, and you want the value of `message` to be `--interactive`. The following command line won't work:
+
+```console
+myapp --interactive
+```
+
+If `myapp` has an `--interactive` option, this input will be interpreted as referring to that option. If the app doesn't have an `--interactive` option, the result will be an unmatched option error.
```

Comment:
This second sentence isn't accurate. If there's no `--interactive` option defined, then this will be parsed as an argument. The `--` escape mechanism is only necessary when there's a matching identifier in the command line app.